code/modules/mob/living/carbon/xenomorph/attack_alien.dm

```diff
@@ -75,13 +75,22 @@
 	if(!(signal_return & COMSIG_XENOMORPH_BONUS_APPLIED))
 		damage_mod += dam_bonus
 
+	var/armor_pen = X.xeno_caste.melee_ap
+
+	if(HAS_TRAIT(X, TRAIT_TURRET_HIDDEN)) //sneeky attack
+		//locate() subtypes aswell, whichever the mob has.
+		var/datum/action/ability/xeno_action/stealth/stealth_skill = locate() in X.actions
+		if(stealth_skill.can_sneak_attack)
+			var/datum/action/ability/activable/xeno/hunter_mark/assassin/mark = X.actions_by_path[/datum/action/ability/activable/xeno/hunter_mark/assassin]
+			if(mark?.marked_target == src) //assassin death mark
+				damage *= 2
+
```

Comment:
As before, this is still cursed and needs to be using the signal call at line 72

Reply:
Huh, maybe I did this that way since this stuff was broken till recently, not sure though this pr take so long i forgot my reasons of doing things certain ways but I guess i'll try it. Maybe I made it check hidden status for modularity of the stealth skill but most likely just cause im dumb, idk.

Reply:
Yeah that might be it. Someone fucked up some xeno sig stuff for ages, so hunter stealth attack bonuses just weren't working at all. I only found and fixed it recently